tests/UnitTests/BankTransactionUnitTest.php

```diff
@@ -30,4 +34,131 @@ public function testCanGetReferenceFromLine()
+
+        self::assertLinesInOrderBasedOnDescription($bank_transaction, $total_line_count);
+    }
+
+    private static function assertLinesInOrderBasedOnDescription(
+        BankTransaction $bank_transaction,
+        int $expected_line_count
+    ) {
+        $i = 0;
+        /** @var Base $line */
+        foreach ($bank_transaction->getLines() as $line) {
+            self::assertSame(
+                $i,
+                (int)$line->getDescription(),
+                'Expected line #' . $i . ', but got line #' . $line->getDescription() .
+                ' of type ' . $line->getLineType()->getValue() . '. Wrong order!'
+            );
+            $i++;
+        }
+
+        self::assertSame($expected_line_count, $i, 'Number of returned lines does not match number of added lines');
+    }
+
+    private function addTotalLineWithDescription(BankTransaction $bankTransaction, string $description): Total
+    {
+        $line = new Total();
+        $line->setDescription($description);
+        $line->setValue(Money::EUR(0));
+
+        $bankTransaction->addLine($line);
+
+        return $line;
+    }
+
+    private function addDetailLineWithDescription(BankTransaction $bankTransaction, string $description): Detail
+    {
+        $line = new Detail();
+        $line->setDescription($description);
+        $line->setValue(Money::EUR(0));
+
+        $bankTransaction->addLine($line);
+
+        return $line;
+    }
+
+    public function testGetLinesReturnsLinesInTheSameOrderAsTheyWereAdded()
+    {
+        $total_line_count = 300;
+
+        $bank_transaction = new BankTransaction();
+
+        $this->addTotalLineWithDescription($bank_transaction, '0');
+
+        for ($i = 1; $i < $total_line_count; $i++) {
+            $this->addDetailLineWithDescription($bank_transaction, (string)$i);
+        }
+
+        self::assertLinesInOrderBasedOnDescription($bank_transaction, $total_line_count);
+    }
+
+    public function testGetLinesReturnsEmptyArrayWhenNoLinesSet()
+    {
+        $bank_transaction = new BankTransaction();
+
+        self::assertCount(0, $bank_transaction->getLines());
+    }
+
+    public function testGetLinesReturnsLinesInSameOrderWhenOnlyDetailsAdded()
+    {
+        $bank_transaction = new BankTransaction();
+
+        $this->addDetailLineWithDescription($bank_transaction, '0');
+        $this->addDetailLineWithDescription($bank_transaction, '1');
+        $this->addDetailLineWithDescription($bank_transaction, '2');
+
+        self::assertLinesInOrderBasedOnDescription($bank_transaction, 3);
+    }
+
+    public function testGetLinesReturnsLinesInSameOrderWhenOnlyTotalLineIsAdded()
+    {
+        $bank_transaction = new BankTransaction();
+
+        $this->addTotalLineWithDescription($bank_transaction, '0');
+
+        self::assertLinesInOrderBasedOnDescription($bank_transaction, 1);
+    }
+
+    public function testAdding1TotalLineAnd500DetailLinesToABankTransactionWorks()
```

Comment:
I'm not sure if it's a good idea to test that 501 items is still OK, the docs explicitly mention the limit of 500, so 501 is a bug, not a feature.

Reply:
Actually agree with this. I've actually changed the implementation to now only allow 500 lines (e.g. 1 TOTAL and 499 DETAIL) and changed the tests to verify this.